src/Config/XmlConfigFileLoader.php

```diff
@@ -0,0 +1,220 @@
+<?php declare(strict_types=1);
+
+
+namespace Shopware\Psh\Config;
+
+use DOMElement;
+use DOMXPath;
+use Symfony\Component\Config\Util\XmlUtils;
+
+/**
+ * Load the config data from an xml file
+ */
+class XmlConfigFileLoader extends ConfigFileLoader
+{
+    const NODE_HEADER = 'header';
+
+    const NODE_PLACEHOLDER = 'placeholder';
+
+    const NODE_PLACEHOLDER_DYNAMIC = 'dynamic';
+
+    const NODE_PLACEHOLDER_CONST = 'const';
+
+    const NODE_PATH = 'path';
+
+    const NODE_ENVIRONMENT = 'environment';
+
+    const NODE_TEMPLATE = 'template';
+
+    const NODE_TEMPLATE_SOURCE = 'source';
+
+    const NODE_TEMPLATE_DESTINATION = 'destination';
+
+    /**
+     * @var ConfigBuilder
+     */
+    private $configBuilder;
+
+    /**
+     * @var string
+     */
+    private $applicationRootDirectory;
+
+    /**
+     * @param ConfigBuilder $configBuilder
+     * @param string $applicationRootDirectory
+     */
+    public function __construct(ConfigBuilder $configBuilder, string $applicationRootDirectory)
+    {
+        $this->configBuilder = $configBuilder;
+        $this->applicationRootDirectory = $applicationRootDirectory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isSupported(string $file): bool
+    {
+        return in_array(pathinfo($file, PATHINFO_BASENAME), ['.psh.xml', '.psh.xml.dist', '.psh.xml.override'], true);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function load(string $file, array $params): Config
+    {
+        $pshConfigNode = $this->loadXmlRoot($file);
+        $this->configBuilder->start();
+
+        try {
+            $this->configBuilder->setHeader(
+                $this->extractNode(self::NODE_HEADER, $pshConfigNode)->nodeValue
+            );
+        } catch (\InvalidArgumentException $e) {
+        }
+
+        $this->setConfigData($file, $pshConfigNode);
+
+        $environments = $this->extractNodes(self::NODE_ENVIRONMENT, $pshConfigNode);
+
+        foreach ($environments as $node) {
+            $this->configBuilder->start($node->getAttribute('name'));
+            $this->setConfigData($file, $node);
+        }
+
+        return $this->configBuilder
+            ->create($params);
+    }
+
+    /**
+     * @param string $file
+     * @param array $pshConfigNode
+     */
```

Comment:
Der php doc block kann von mir aus weg. Steht ja alles schon in der Methoden Signatur drin :-)